### ISSUE TITLE
[UX] Fix page titles for the various "Add xyz" tasks under "Structure"

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1456]
- The menu add page (`admin/structure/menu/add`) has "Menus" as title - should be something like "Add menu".
- The menu link add page (`admin/structure/menu/manage/[menu-name]/add`) has a "[Menu name]" title - should be something like "[Menu name]: Add link". That page has two inactive tabs: "List links" and "Edit menu". We should lose these from the add link page to be in tandem with the link edit page. The way it is now is weird:

![backdrop-menu_link_add_page](https://cloud.githubusercontent.com/assets/2423362/12009276/960e4c42-acc5-11e5-9690-b69553cd8af1.png)

The menu link edit page does not have these two tabs and the page title is ok ("Edit menu link"):

![backdrop-menu_link_edit_page](https://cloud.githubusercontent.com/assets/2423362/12009288/e11fbcc0-acc5-11e5-8be8-8a79b7688ec3.png)

...although, it would be better IMO to have it be "Edit menu link [menu_link_title]" instead in order to differentiate in case one has multiple tabs open in their browser.

---

~PR by @BWPanda: https://github.com/backdrop/backdrop/pull/2088~
PR by @quicksketch: https://github.com/backdrop/backdrop/pull/2097

**Comment by klonos** [comment]
I often open many edit/add link tabs in my browser and it is really hard to find the right one. This is annoying. Can we set a milestone please?

**Comment by BWPanda** [comment]
Simple PR fixes this.

**Comment by herbdool** [comment]
The PR fixes the main issue by getting it to display Add Link as title and gets rid of the tabs.

**Comment by quicksketch** [comment]
I filed a new PR at https://github.com/backdrop/backdrop/pull/2097 that should fix the display of action links throughout Backdrop, and the titles for adding a menu or menu link specifically.

The first fix is simple: let `menu_get_active_title()` return the action link title instead of using the parent item title.

However, after fixing that, it appears as though we have some cruft from the Drupal 6 era in menu module that manually creates individual menu links for each created menu. I couldn't see any reason for this, as we don't do this with any other menu links that contain wildcards (e.g. editing content types, vocabularies, layouts, views, etc.) I think it's simply code that had been ported forward because no one really understood what it did.

So this PR removes that cruft (the entire `menu_enable()` function), cleans up the database menu_links table, and also switches the title that is used when on a page created by a local action link.

**Comment by olafgrabienski** [comment]
Good catch with the disappeared action links! The sandbox site of the new PR works as expected: You can add menus and links again using the action links; the title of these pages are "Add menu" and "Add link" respectively.

Compared to the former PR, the two inactive tabs on the *Add link* page are still visible.

**Comment by quicksketch** [comment]
> Compared to the former PR, the two inactive tabs on the Add link page are still visible.

If we're going to change that, we'd need to come up with something global. This happens for all pages where there are both local tasks (tabs) and local actions (the "+ Add [whatever]" links). You can see this same thing on `admin/structure/layouts/add`, `admin/structure/views/add`, `admin/config/urls/path/add`, `admin/config/urls/redirect/add`, and probably other places. So perhaps we save that for a different issue.

**Comment by quicksketch** [comment]
I updated the PR a bit to also remove unnecessary `backdrop_set_title()` calls elsewhere. Now that action links use their own link title as the page title, we can remove calls to `backdrop_set_title()` when adding a block, text editor, layout, block, or view.

**Comment by BWPanda** [comment]
Closedmy PR (https://github.com/backdrop/backdrop/pull/2088) in favour of https://github.com/backdrop/backdrop/pull/2097

**Comment by klonos** [comment]
...this needs a rebase and some conflicts resolution.

**Comment by klonos** [comment]
Bumped into this again while working on #6578, where the page title for `admin/structure/file-types/add` is "File types" instead of "Add file type" 👎🏼 ...and is showing the horizontal tabs 👎🏼 :
<img width="411" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/5a963085-8607-49ae-99f8-00a3908441ad">

I'm looping back to this to get it out of the way.

**Comment by klonos** [comment]
> ...and is showing the horizontal tabs 👎🏼 :

That is intentional after all - I just thought that it was random because I couldn't figure out the logic. So lets address just the titles issue here, and decide if hiding the tabs is a good thing or a bad thing UX-wise in a separate issue: #6579

**Comment by klonos** [comment]
OK, PR ready for review: https://github.com/backdrop/backdrop/pull/4770

**Comment by stpaultim** [comment]
@klonos This needs a rebase. No available sandbox.

**Comment by laryn** [comment]
@stpaultim I rebased it.
